src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.MenuItems.cs

```diff
@@ -887,8 +895,27 @@ private void StateHistorySettingsMenuItem_Click(object sender, EventArgs e)
 		{
 			using GreenzoneSettings form = new(
 				DialogController,
-				new ZwinderStateManagerSettings(CurrentTasMovie.TasStateManager.Settings),
-				(s, k) => { CurrentTasMovie.TasStateManager.UpdateSettings(s, k); },
+				CurrentTasMovie.TasStateManager.Settings.Clone(),
+				(s, k) => { CurrentTasMovie.TasStateManager = CurrentTasMovie.TasStateManager.UpdateSettings(s, k); },

```

Comment:
This weird construct suggests to me that you're doing something wrong, but I couldn't say what.